File: java-samples/Features/queryablestate-carinsurance/workflows/src/test/java/net/corda/samples/carinsurance/flows/FlowTests.java
package net.corda.samples.carinsurance.flows;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.transactions.SignedTransaction;
import net.corda.testing.node.*;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters(ImmutableList.of(
            TestCordapp.findCordapp("net.corda.samples.carinsurance.contracts"),
            TestCordapp.findCordapp("net.corda.samples.carinsurance.flows")
    )).withNotarySpecs(ImmutableList.of(new MockNetworkNotarySpec(CordaX500Name.parse("O=Notary,L=London,C=GB"))))
    );
    private final StartedMockNode a = network.createNode();
    private final StartedMockNode b = network.createNode();


    @Before
    public void setup() {
        network.runNetwork();
    }

    @After
    public void tearDown() {
        network.stopNodes();
    }

    //simple example test to test if the issue insurance flow only carries one output.
    @Test
    public void issueInsuranceFlowTest() throws Exception {
        VehicleInfo car = new VehicleInfo(
                "I4U64FY56I48Y",
                "165421658465465",
                "BMW",
                "M3",
                "MPower",
                "Black",
                "gas");
        InsuranceInfo policy1 = new InsuranceInfo(
                "8742",
                2000,
                18,
                49,
                car);

        IssueInsuranceFlow.IssueInsuranceInitiator flow = new IssueInsuranceFlow.IssueInsuranceInitiator(policy1, b.getInfo().getLegalIdentities().get(0));
        CordaFuture<SignedTransaction> future = a.startFlow(flow);
        network.runNetwork();
        SignedTransaction ptx = future.get();

        //assertion for single output
        assertEquals(1, ptx.getTx().getOutputStates().size());

    }
}

File: kotlin-samples/Features/queryableState-carinsurance/workflows/src/test/kotlin/net/corda/samples/carinsurance/flows/FlowTests.kt
package net.corda.samples.carinsurance.flows

import com.google.common.collect.ImmutableList
import net.corda.core.identity.CordaX500Name
import net.corda.core.transactions.SignedTransaction
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkNotarySpec
import net.corda.testing.node.MockNetworkParameters
import net.corda.testing.node.TestCordapp
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class FlowTests {
    private val network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = ImmutableList.of(
            TestCordapp.findCordapp("net.corda.samples.carinsurance.contracts"),
            TestCordapp.findCordapp("net.corda.samples.carinsurance.flows")
    ),
            notarySpecs = listOf(MockNetworkNotarySpec(CordaX500Name("Notary","London","GB")))
    ))
    private val a = network.createNode()
    private val b = network.createNode()

    @Before
    fun setup() {
        network.runNetwork()
    }

    @After
    fun tearDown() {
        network.stopNodes()
    }

    //simple example test to test if the issue insurance flow only carries one output.
    @Test
    @Throws(Exception::class)
    fun issueInsuranceFlowTest() {

        val car = VehicleInfo(
                "I4U64FY56I48Y",
                "165421658465465",
                "BMW",
                "M3",
                "MPower",
                "Black",
                "gas")

        val policy1 = InsuranceInfo(
                "8742",
                2000,
                18,
                49,
                car)

        val flow: IssueInsurance = IssueInsurance(policy1, b.info.legalIdentities[0])
        val future = a.startFlow<SignedTransaction>(flow)
        network.runNetwork()
        val ptx = future.get()

        //assertion for single output
        Assert.assertEquals(1, ptx.tx.outputStates.size.toLong())
    }
}
